code/app/src/main/java/com/example/bread/repository/ParticipantRepository.java

```diff
@@ -115,4 +163,73 @@ public void checkIfAlreadyFollowed(@NonNull String username, String followerUser
 
     public void checkIfAlreadyFollowing(@NonNull String username, String followingUsername, @NonNull OnSuccessListener<Boolean> onSuccessListener, OnFailureListener onFailureListener) {
     }
+
+    public void acceptFollowRequest(String requestId, String senderId, String receiverId, OnSuccessListener<Void> onSuccessListener, OnFailureListener onFailureListener) {
+        WriteBatch batch = db.batch();
+
+        DocumentReference senderRef = db.collection("participants").document(senderId);
```

Comment:
documents are uniquely defined by the username not the Uid, this won't work with the current setup at the moment.